Feat/chattinguser

maindishes opened the pull request:
- 채팅방생성 1. 비번방 , 2. 공개방
- 채팅방 입장 요청 POST 시 
  -  비번방 에 들어갈때
     - 해쉬화된 패스워드 compare  확인후 입장 여부 . ( 추후 밴리스트 체크후 들어가져야됨)
  -  공개방에 들어갈때 
     - 그냥 들어가짐 ( 추후 밴리스트 체크후 들어가져야됨)
 -----
# TODO
- 비번 일치, 공개방 입장시 소캣 연결 (join 채팅방소켓)
  - 채팅 소켓 연결 
- 밴리스트 체크후 들어가져야됨

maindishes committed:
feat: 채널방 비밀번호 해쉬화

maindishes committed:
feat: 채널방 비밀번호 해쉬화

- Post Body로 비밀 번호 요청 왔을때 본래 방 비밀번호와 비교

maindishes committed:
fix : private 데이터 smallint -> 그냥 boolean으로 나중에 다시 최적화

maindishes committed:
fix: password , owner 값들이 빈값들어 올경우 예외처리

maindishes committed:
feat: 비밀번호 있을때 , 없을때 예외처리

- 비밀 번호 있을때 확인후 private = true 로 해주는것으로 변경
- 프론트에서 주는 값에 대한 의존성 해제

maindishes committed:
ERROR: Do not check this code

maindishes committed:
Feat : 채팅방 입장시공개방 , 비공개방 함수 나눠서 예외 처리

maindishes committed:
Merge branch 'develop' of https://github.com/42-TRANSENDENCE/backend into feat/chattinguser

minjune8506 commented on src/channels/channels.entity.ts:
```diff
@@ -17,14 +18,15 @@ export class Channels {
-      width: 1,
       default: () => "'0'",
     })
     private: boolean | null;
```
이거 Enum으로 상태 관리하는건 어때요
ChannelType 이런식으로해서 PRIVATE / PUBLIC / PROTECTED

maindishes replied:
옙 나중에 어차피 DM기능 추가 되면 해야해서 Enum으로 해야 합니다. 


> 이거 지금 테스트하려고 올리는건가요??

아니요 , 최소? 단위로 커밋 해놓으려고 한거입니다. 머지 되면 좋긴 한데 이번 주말 쯤에 소켓 연결 한거 한번에 머지 해도 상관 없습니다~

minjune8506 replied:
넹 알게씁니다아아

minjune8506 replied:
제거는 리뷰한번 해주세여 올리고 건우한테 테스트해보라고 할려구여

maindishes replied:
지금 보는중 입니당 으렵네요 일단 테스트를 위해 login part 머지 꼬우

minjune8506 reviewed (commented):
이거 지금 테스트하려고 올리는건가요??

maindishes committed:
FEAT : 방생성시 방 객체 프론트에 socket으로 보내기

maindishes committed:
Chore: remove useless comment

maindishes committed:
Feat: chats api 수정

maindishes committed:
Feat : chat API 수정

- api url 수정

maindishes committed:
Fix : 이름 수정

maindishes committed:
Chore : 이름 수정

maindishes committed:
Chore: 이름 수정후 import 부분 자동 수정

maindishes committed:
Feat: status 200 으로 수정 하는 부분 Promise 로 해결

- ChannelMember 에 유저가 있으면 db에 추가 하는 부분 예외처리

maindishes committed:
Chore: Channels 모듈 추가

maindishes committed:
Feat: db 구현중... Channel, Chats, ChannelMember

maindishes committed:
Feat: 일단 채팅 요청 하나 할때마다 저장 하는걸로 구현-> 추후 다른방법 생각

maindishes committed:
Feat: join-room 이벤트 구현, leave-room event 구현중..

maindishes committed:
Chore: package

maindishes committed:
merge

maindishes committed:
Chore : 채팅방정보[[json 형태의 채널 멤버],[],1] #60

- 형태로 반환

maindishes committed:
Feat : 채팅방 정보 요청시 존재하지 않을때 예외처리 #60

maindishes committed:
Feat : Owner 가 채널의 admin 권한 주기 , db에 admin col 추가 #61

maindishes committed:
Fix: 방입장 요청시 인자  createDto > password:string  으로변경

maindishes committed:
Fix : circular dependency 해결 #63

maindishes committed:
Feat: 채팅방 나갈때 DB 에서 삭제 #59

maindishes committed:
Feat: leave-room  이벤트 발생시  socket연결 해제 추가 #59

maindishes committed:
Feat: leave-room Event 예외 처리 #70

maindishes committed:
Feat: add db BanMember, muteMember #72

- BanMember, MuteMember entity 생성
- IsBanned 함수 호출할때 시간 지나있으면 BanMember에서 삭제
- BanMember 에 있으면 채팅방 입장 제한

maindishes committed:
Feat : channel Service 의 함수를 Chat Service에서 쓰기#72

maindishes committed:
Feat: IsMutted 함수 구현 #72

maindishes committed:
Feat: MuteMember에 있으면 채팅 저장 하지 않기 #72

- 소켓 관련해서는 구현중 ...
- TODO: 채팅창 연결해서 User.id랑 연결해서 테스트
- TODO: 인자들 정리, entity 도 정리

maindishes committed:
chore: 주석

maindishes committed:
chore: 파일 구조 channel 하위 로 옮김

maindishes committed:
Mrege: before merge to dev

maindishes committed:
Merge

maindishes committed:
Merge branch 'develop' of https://github.com/42-TRANSENDENCE/backend into feat/chattinguser

maindishes committed:
Feat: mutelist 캐시로 관리하기

maindishes committed:
Chore : delete unused modules

maindishes committed:
Chore: 전 커밋 이슈랑 연결을 위한 커밋 #78

maindishes committed:
Feat: 초당 요청횟수 제한

minjune8506 commented on src/ratelimit/rateLimitMiddleware.ts:
```diff
@@ -0,0 +1,27 @@
+      next();
+    } catch (err) {
+      // The user has exceeded the request rate, send a 429 response
+      res.status(429).send('Too Many Requests');
```
429 보내면 front에서 계속 다시 요청하는 방향으로 하나요?

maindishes replied:
아니요 일단 모든 route 에 적용되는지 해보려고 한거고 다시 요청하는 방향으로 하는건 아닙니당 원하면 해도 되구여 글고 초당 몇번으로 제한할건지도 생각해 봐야합니당

minjune8506 replied:
API 제한해야하는게 42 API를 이용하는 과정에서 1초에 2번 제한이 있어서 사용하는거에요
그래서 42 API 요청하는 것만 1초에 2번으로 제한하면 되고 나머지는 필요 없씁니다.

minjune8506 replied:
근데 이거 다시보니까 한 IP당 2번 요청제한하는거 같은데요?

maindishes replied:
아 맞네  그럼 초당 api 함수 호출 횟수를 제한 해야하는거져? 로그인만 2회로 
- 그리고ip당 초당 몇회로 제한 해야하지않아? 사람인지 봇인지 확인하는게 그거 아닌가여?
- 그리고  그 요청 개많이 해서 서버 터지게 하는거 디도스? 는 초당 api 함수 제한 하게 하면 되는건가

minjune8506 replied:
네 디도스 막으려면 그렇게 하긴 해야할듯합뉘다

minjune8506 replied:
@maindishes 형 이거 머지할거면.. 브랜치 하나파서 cherry-pick으로 정리하고 하나씩 올려줘요..

이거 절대 리뷰못함

maindishes replied:
어케이 아니 기능 하나만 더하고 풀리퀘 날려야지 하다 요지경 됨  cherry-pick 으로 할게유

maindishes committed:
Merge branch 'develop' of https://github.com/42-TRANSENDENCE/backend into feat/chattinguser

maindishes committed:
Merge: pull feat/chattinguser to devlop
